### PR TITLE
[Backend] TRT cast GPU input from int64 to int32, output from int32 to int64, and Windows support building CUDA files

### PR DESCRIPTION
**Pull request by wang-xinyu**
### PR types(PR类型)

Backend

### Describe

- Add a CUDA cast function, implemented with CUDA kernel
- TRT backend: cast GPU input int64 to int32
- Verified on text/uie/cpp demo, the inference result of GPU int64 input is same as CPU int64 input
- Windows cmake changes to support building CUDA files
- TRT backend: cast GPU output int32 to int64 when onnx output is int64, verified on a custom onnx model

### CODE REVIEW

File: fastdeploy/backends/tensorrt/trt_backend.cc [hunk 1]
```diff
@@ -234,6 +236,7 @@ bool TrtBackend::InitFromOnnx(const std::string& model_file,
     inputs_desc_[i].name = name;
     inputs_desc_[i].shape.assign(shape.begin(), shape.end());
     inputs_desc_[i].dtype = ReaderDtypeToTrtDtype(onnx_reader.inputs[i].dtype);
+    inputs_desc_[i].original_dtype = ReaderDtypeToFDDtype(onnx_reader.inputs[i].dtype);
```
Comment (jiangjiajun) [inline fastdeploy/backends/tensorrt/trt_backend.cc:239]: GetInputInfo(), GetOutputInfo()应该也是返回真实的dtype，当前仍然会返回`int32`

File: fastdeploy/backends/tensorrt/trt_backend.cc [hunk 2]
```diff
@@ -358,11 +397,15 @@ void TrtBackend::SetInputs(const std::vector<FDTensor>& inputs) {
 
     if (item.device == Device::GPU) {
       if (item.dtype == FDDataType::INT64) {
-        // TODO(liqi): cast int64 to int32
-        // TRT don't support INT64
-        FDASSERT(false,
-                 "TRT don't support INT64 input on GPU, "
-                 "please use INT32 input");
+        FDWARNING << "TRT doesn't support INT64 input, the input tensor will be "
+                     "casted to INT32" << std::endl;
```
Comment (jiangjiajun) [inline fastdeploy/backends/tensorrt/trt_backend.cc]: 这行WARNING可以删除，看起来是每次Infer都会输出出来

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

File: fastdeploy/backends/tensorrt/trt_backend.cc (hunk 1 above)
Comment (wang-xinyu) [reply fastdeploy/backends/tensorrt/trt_backend.cc:239]: fixed

File: fastdeploy/backends/tensorrt/trt_backend.cc (hunk 2 above)
Comment (wang-xinyu) [reply fastdeploy/backends/tensorrt/trt_backend.cc]: deleted

**Review by jiangjiajun — APPROVED** [review]

**Pull request merged**